doc/languages-frameworks/rust.section.md

```diff
@@ -122,16 +122,38 @@ rustPlatform.buildRustPackage rec {
 
   cargoLock = {
     lockFile = ./Cargo.lock;
-  }
+  };
 
   # ...
 }
 ```
 
 This will retrieve the dependencies using fixed-output derivations from
-the specified lockfile. Note that setting `cargoLock.lockFile` doesn't
-add a `Cargo.lock` to your `src`, and a `Cargo.lock` is still required
-to build a rust package. A simple fix is to use:
+the specified lockfile.
+
+Alternatively, `cargoLock.lockFileContents` can be set to a string of
+the contents of a `Cargo.lock` file, for example if you need to
```

Comment:
I would add in brackets after file something like
`(in case the src Cargo.lock needs patching. This cannot be done in patches, since the patches run after the fetching of the dependencies)`

Reply:
Good point. I rephrased the entire paragraph to hopefully convey the purpose more clearly.